### PR TITLE
[JUJU-1369] Make juju-wait-for a standard command instead of a plugin

### PR DESCRIPTION
**Pull request by manadart**
Moves `juju wait-for` from being a plugin to being a standard Juju command.

We also remove the make jobs and snapcraft logic for the plugin. We will need to update the release jobs to eschew packing what was previously a stand-alone plugin for the command.

## QA steps

- Bootstrap and add a model.
- `juju help wait-for`.
- `juju help wait-for application`.
- `juju wait-for application mysql --query='name=="mysql" && (status=="active" || status=="idle")'`
- In another terminal: `juju deploy mysql --series xenial`
- See the `wait-for` call return once the deployment quiesces.

## Documentation changes

There are probably references to it.

## Bug reference

N/A

### CODE REVIEW

File: cmd/juju/waitfor/waitfor.go [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+var logger = loggo.GetLogger("juju.plugins.waitfor")
+
+var waitForDoc = `
+Juju wait-for attempts to wait for a given entity to reach a goal state.
+`
+
+// NewWaitForCommand creates the wait-for supercommand and registers the
+// subcommands that it supports.
+func NewWaitForCommand() cmd.Command {
+	waitFor := cmd.NewSuperCommand(cmd.SuperCommandParams{
```
Comment (manadart) [inline cmd/juju/waitfor/waitfor.go:22]: So this doesn't work. I'll have to reorganise how the sub-commands work here.

File: cmd/juju/waitfor/waitfor.go [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+
+	_ "github.com/juju/juju/provider/all"
+)
+
+var logger = loggo.GetLogger("juju.plugins.waitfor")
+
+var waitForDoc = `
+Waits for a specified model, machine, application or unit to reach a state
+defined by the supplied query.
+`
```
Comment (juanmanuel-tirado) [inline cmd/juju/waitfor/waitfor.go:18]: If wold be nice to provide a minimal example here.

**Review by juanmanuel-tirado — APPROVED** [review]
Just a minor suggestion about the docs for the command.

LG2M

**Pull request merged**